config/http-headers.js

```diff
@@ -14,8 +14,8 @@ let matrixMessengerEmbed = '';
 let matrixMessengerHomeserver = '';
 let matrixMessengerDiscoverDomain = '';
 if (Configuration.get('FEATURE_MATRIX_MESSENGER_ENABLED')) {
-	matrixMessengerEmbed = Configuration.get('MATRIX_MESSENGER_EMBED_URI');
-	matrixMessengerHomeserver = Configuration.get('MATRIX_MESSENGER_HOMESERVER_URI');
+	matrixMessengerEmbed = Configuration.get('MATRIX_MESSENGER__EMBED_URI');
+	matrixMessengerHomeserver = Configuration.get('MATRIX_MESSENGER__URI');
 	matrixMessengerDiscoverDomain = matrixMessengerHomeserver.replace('matrix.', '');
```

Comment:
The discoveryDomain should not be generated, let's add a configuration variable for it.